sirepo/job_supervisor.py

```diff
@@ -738,11 +745,14 @@ async def _receive_api_runStatus(self, req):
         r = self._status_reply(req)
         if r:
             return r
-        return await self._send_with_single_reply(
+        r = await self._send_with_single_reply(
             job.OP_ANALYSIS,
             req,
             jobCmd='sequential_result',
         )
+        if r.state == job.ERROR:
```

Comment:
is this the reason always for an error. Could it be something other than an init_db?

Reply:
I believe any time we do a runStatus and get an error back from the agent then we want to init_db. If we get back error from the agent then we can't read the results so the compute needs to be run again. There are some edge cases (ex fastcgi socket dies) where we might get back error but we could try again and maybe read the results. It isn't great to recompute in those cases but trickier to decipher them from an error reading the actual results. 

I interpreted [your comment here](https://github.com/radiasoft/sirepo/issues/2697#issuecomment-664667382) to mean what I implemented. Did I get it right?

Reply:
I think if r.state error then recompute is fine.